bin/opam.ml

```diff
@@ -267,6 +267,17 @@ let get_pkgs ?build_dir ?opam ?distrib_file ?readme ?change_log ?publish_msg
     Pkg.distrib_file ~dry_run pkg
   in
   Pkg.infer_pkg_names Fpath.(v ".") pkg_names >>= fun pkg_names ->
+  (* shuffle main package to front if it exists *)
```

Comment:
How about keeping a `string` and move that logic when constructing the record? if the main package can be detected, use it; otherwise set `main_pkg` to `List.hd pkgs`.

Reply:
No I meant the opposite: move this logic to the `Pkg` module (which we still need to rename to `Repo`) and have a function there return the record.

The logic for the opam command would use that and it could then be used elswhere as well.

Might be worth looking at `Undraft` as well which has a lot of code copied from `Opam`. I think it wants to know the main pkg as well.